File: 0844-backspace-string-compare/0844-backspace-string-compare.cpp
class Solution {
public:
    bool backspaceCompare(string s, string t) {
        stack<char>ss;
        stack<char>st;
        for(int i=0;i<s.length();i++){
            if(s[i]!='#'){
                ss.push(s[i]);
            }
            else{
                if(!ss.empty()){
                    ss.pop();
                } 
            }
        }
        for(int i=0;i<t.length();i++){
            if(t[i]!='#'){
                st.push(t[i]);
            }
            else{
                if(!st.empty()){
                    st.pop();
                }   
            }
        }
        if(ss==st){
            return true;
        }
        else{
            return false;
        }   
    }
};

File: 0844-backspace-string-compare/0844-backspace-string-compare.java
class Solution {
    public boolean backspaceCompare(String s, String t) {
        Stack<Character> sChar = new Stack<>();
        Stack<Character> tChar = new Stack<>();
        
        for(int i = 0; i<s.length(); i++){
            if(s.charAt(i) != '#'){
                sChar.push(s.charAt(i));
            }
            else{
                if(!sChar.isEmpty()){
                    sChar.pop();
                }   
            }
        }
        
        for(int i = 0; i<t.length(); i++){
            if(t.charAt(i) != '#'){
                tChar.push(t.charAt(i));
            }
            else{
                if(!tChar.isEmpty()){
                    tChar.pop();
                }
            }
        }
        
        return sChar.equals(tChar);
    }
}